src/BlockTemplatesController.php

```diff
@@ -53,10 +53,37 @@ public function add_block_templates( $query_result, $query, $template_type ) {
 			return $query_result;
 		}
 
+		$post_type      = isset( $query['post_type'] ) ? $query['post_type'] : '';
 		$template_files = $this->get_block_templates();
 
+		// @todo: Add apply_filters to _gutenberg_get_template_files() in Gutenberg to prevent duplication of logic.
 		foreach ( $template_files as $template_file ) {
-			$query_result[] = BlockTemplateUtils::gutenberg_build_template_result_from_file( $template_file, 'wp_template' );
+			$template = BlockTemplateUtils::gutenberg_build_template_result_from_file( $template_file, 'wp_template' );
+
+			if ( $post_type && ! $template->is_custom ) {
+				continue;
+			}
```

Comment:
@tjcafferkey the `$template` object returned by `BlockTemplateUtils::gutenberg_build_template_result_from_file()` doesn't have `is_custom` property on WP 5.8.1. The `WP_Block_Template` class shipped with 5.8.1 doesn't have that property (the one in Gutenberg does) so I'm getting this warning:

```
Warning: Undefined property: WP_Block_Template::$is_custom in /wp-content/plugins/woocommerce-gutenberg-products-block/src/BlockTemplatesController.php on line 63
```

Reply:
Thanks @dinhtungdu. I will open a PR to fix this by setting the `is_custom` value to `false` in `BlockTemplateUtils::gutenberg_build_template_result_from_file()` seeing as we _know_ this is being loaded from the filesystem so it should be `false`.

`is_custom` templates are ones that have been edited and are being retrieved from the DB.

Reply:
Resolved in #5067